services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -0,0 +1,324 @@
+                    status.HTTP_500_INTERNAL_SERVER_ERROR, detail=err.message
+                ) from err
+
+        return results
+
+
+@containers_router.get(
+    "/containers/{id}:logs",
+    responses={
+        status.HTTP_500_INTERNAL_SERVER_ERROR: {
+            "description": "Container does not exists"
+        }
+    },
+)
+async def get_container_logs(
+    id: str,
+    since: int = Query(
+        0,
+        title="Timestamp",
+        description="Only return logs since this time, as a UNIX timestamp",
+    ),
+    until: int = Query(
+        0,
+        title="Timestamp",
+        description="Only return logs before this time, as a UNIX timestamp",
+    ),
+    timestamps: bool = Query(
+        False,
+        title="Display timestamps",
+        description="Enabling this parameter will include timestamps in logs",
+    ),
+    shared_store: SharedStore = Depends(get_shared_store),
+) -> Union[str, Dict[str, Any]]:
+    """ Returns the logs of a given container if found """
+    # TODO: remove from here and dump directly into the logs of this service
+    # do this in PR#1887
+    if id not in shared_store.container_names:
+        message = f"No container '{id}' was started. Started containers '{shared_store.container_names}'"
+        logger.warning(message)
+        raise HTTPException(status.HTTP_500_INTERNAL_SERVER_ERROR, detail=message)
+
+    with docker_client() as docker:
+        try:
+            container_instance = await docker.containers.get(id)
+
+            args = dict(stdout=True, stderr=True)
+            if timestamps:
+                args["timestamps"] = True
+
+            if since or until:
+                raise HTTPException(
+                    status.HTTP_501_NOT_IMPLEMENTED,
+                    detail="since and until options are still not implemented",
+                )
+
+            container_logs: str = await container_instance.log(**args)
+            return container_logs
+        except aiodocker.exceptions.DockerError as err:
+            logger.warning(
+                "An unexpected Docker error occurred:\n%s", str(traceback.format_exc())
+            )
+            raise HTTPException(
+                status.HTTP_500_INTERNAL_SERVER_ERROR, detail=err.message
+            ) from err
+
+
+@containers_router.get(
+    "/containers/{id}:inspect",
+    responses={
+        status.HTTP_500_INTERNAL_SERVER_ERROR: {
+            "description": "Container does not exist"
+        }
+    },
+)
+async def inspect_container(
+    id: str, shared_store: SharedStore = Depends(get_shared_store)
+) -> Dict[str, Any]:
+    """ Returns information about the container, like docker inspect command """
+
+    if id not in shared_store.container_names:
+        message = f"No container '{id}' was started. Started containers '{shared_store.container_names}'"
+        logger.warning(message)
+        raise HTTPException(status.HTTP_500_INTERNAL_SERVER_ERROR, detail=message)
```

Comment:
Take into account that if here I would raise any other non-http exception, it gets automatically translated in 500
therefore, i would always try to generate any other status-code if possible. here, againg I think is a 404.